experimental/dds/tree2/src/shared-tree-core/editManager.ts

```diff
@@ -240,6 +328,18 @@ export class EditManager<
 			trunkTailSequenceId = minSequenceId(trunkTailSequenceId, minimumBranchBaseSequenceId);
 		}
 
+		// TODO get the oldest revertible sequence id from all registered branches
+		if (this.oldestRevertibleSequenceId !== undefined) {
+			// use a smaller sequence number so that the oldest revertible is not trimmed
```

Comment:
Are you sure that you need to do this (the -1)? Of all commits that get evicted, the newest one is special in that it doesn't actually get dropped but becomes the `trunkBase` (see line 360). This means that after eviction, commits which were branching off of it can still rebase over it. It's still there (even though it is not counted as part of the trunk). So I would expect it to be okay if you used the `oldestRevertibleSequenceId` directly, since the invariant you're trying to uphold is that the oldest revertible can be rebased over (not necessarily that the oldest revertible remains "in the trunk").

Reply:
The problem is that when we make it the trunk base, we also get rid of its change so there's nothing for us to revert.

Reply:
Haha oh darn, I guess you need that too. Why not just do the really obscure and confusing thing, and delete the code that overwrites the new trunk base's change? After all, it says it's "not strictly necessary" :) :)